ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatInputStream.java

```diff
@@ -98,6 +98,11 @@ public int read(byte b[], int off, int len) throws IOException {
     return totalRead;
   }
 
+  @Override
+  public int available() {
```

Comment:
This method is overridden because `channel.read()` depends on it. 

If default implementation is used, `channel.read()` can read 8k every time, the change in `appendFrom` won't have any improvement(still 8k write).

Reply:
Will chat with you to understand this comment